### PR TITLE
Fill the OTel part of the flare, traverse config sources

### PR DESCRIPTION
**Pull request by dustmop**
### What does this PR do?

An earlier PR put the FlareProvider on the otelcol/collector/impl which is part of the otel-agent. Instead it should be part of the core-agent, in otelcol/collector/impl-pipeline folder.

Fill in the flare details by contacting the OTel extension and requesting config info from it. For any additional sources, crawl their responses and add it all to the flare.

### Motivation

https://docs.google.com/document/d/1EirN3XR2YYfqYmyl8DrPGZC6d4rG-Y00lRVc4Pv3BHQ/edit

### Describe how to test/QA your changes

Basic unit test exists to test the flare filler, more coming as implementation is moved further along.

### CODE REVIEW

File: comp/otelcol/collector/impl-pipeline/pipeline.go [hunk 1]
```diff
@@ -112,6 +122,125 @@ func (c *collectorImpl) stop(context.Context) error {
 	return nil
 }
 
+func (c *collectorImpl) fillFlare(fb flarebuilder.FlareBuilder) error {
+	otelFlareEnabled := c.config.GetBool("otel-agent.enabled")
```
Comment (songy23) [inline comp/otelcol/collector/impl-pipeline/pipeline.go:126]: Are these 2 configs added? If not maybe add a todo?

**Review by songy23 — COMMENTED** [review]
Need to fix a bunch of lints https://github.com/DataDog/datadog-agent/actions/runs/9390698730/job/25861152026?pr=26395

**Review by CelianR — APPROVED** [review]

**Review by dineshg13 — CHANGES_REQUESTED** [review]
When OTel agent is running we wouldn't enable collector in core-agent. Can we make flare collector part of any other component ?

**Review by dineshg13 — APPROVED** [review]
Discussed offline. We should be good for now. We will revisit if we need to move flareprovider out of collector component in the agent.

**Comment by dustmop** [comment]
That sounds good. Even if the collector component is disabled by the config, fx will still receive the flareProvider so the flare will be built correctly.

File: comp/otelcol/collector/impl-pipeline/pipeline.go [hunk 2]
```diff
@@ -112,6 +123,166 @@ func (c *collectorImpl) stop(context.Context) error {
 	return nil
 }
 
+func (c *collectorImpl) fillFlare(fb flaretypes.FlareBuilder) error {
+	if !c.config.GetBool("otel.enabled") {
+		return nil
```
Comment (hush-hush) [inline comp/otelcol/collector/impl-pipeline/pipeline.go:128]:
```suggestion
		fb.AddFile("otel/otel-agent.log", []byte("'otel.enabled' is disabled in the configuration))
		return nil
```

File: comp/otelcol/collector/impl-pipeline/pipeline.go [hunk 3]
```diff
@@ -112,6 +123,166 @@ func (c *collectorImpl) stop(context.Context) error {
...
+	}
+
+	// request config from Otel-Agent
+	responseBytes, err := c.requestOtelConfigInfo(c.config.GetInt("otel.extension_url"))
+	if err != nil {
+		fb.AddFile("otel/otel-agent.log", []byte(fmt.Sprintf("did not get otel-agent configuration: %v", err))) //nolint:errcheck
```
Comment (hush-hush) [inline comp/otelcol/collector/impl-pipeline/pipeline.go:134]: `errcheck` should ignore the flarebuilders method (see [this](https://github.com/DataDog/datadog-agent/blob/main/.golangci.yml#L102)).

File: comp/otelcol/collector/impl-pipeline/pipeline.go [hunk 4]
```diff
@@ -112,6 +123,166 @@ func (c *collectorImpl) stop(context.Context) error {
...
+	fb.AddFile("otel/otel-flare/startup.cfg", []byte(toJSON(responseInfo.StartupConf)))     //nolint:errcheck
+	fb.AddFile("otel/otel-flare/runtime.cfg", []byte(toJSON(responseInfo.RuntimeConf)))     //nolint:errcheck
+	fb.AddFile("otel/otel-flare/environment.cfg", []byte(toJSON(responseInfo.Environment))) //nolint:errcheck
+	fb.AddFile("otel/otel-flare/cmdline.txt", []byte(responseInfo.Cmdline))                 //nolint:errcheck
+
+	// retrieve each source of configuration
+	for _, src := range responseInfo.Sources {
+		var err error
+		var response *http.Response
+		retries := 3
```
Comment (hush-hush) [inline comp/otelcol/collector/impl-pipeline/pipeline.go:155]: Why are we expecting error from the OTEL endpoint ?

**Review by hush-hush — CHANGES_REQUESTED** [review]

**Review by truthbk — APPROVED** [review]

File: comp/otelcol/collector/impl-pipeline/pipeline.go (hunk 3 above)
Comment (dustmop) [reply comp/otelcol/collector/impl-pipeline/pipeline.go]: Ah I see. I had, in an earlier PR, split flare/types into a non-fx (compdef) version, which this linter doesn't cover. I'm now adding a rule to acknowledge these errchecks as well.

File: comp/otelcol/collector/impl-pipeline/pipeline.go (hunk 4 above)
Comment (dustmop) [reply comp/otelcol/collector/impl-pipeline/pipeline.go]: To be honest it was an attempt to handle the flakiness in this test in the CI, but that doesn't seem to be fixing it. I have removed it.

**Review by hush-hush — APPROVED** [review]

**Pull request merged**